src/main/java/dev/codesoapbox/dummy4j/ConvenienceMethods.java

```diff
@@ -53,22 +53,32 @@ public ConvenienceMethods(RandomService random) {
      * Returns a random element from an array
      *
      * @param array the array to pick from
-     * @param <T> the type of object to return
+     * @param <T>   the type of object to return
      * @return a random element
      */
     @SafeVarargs
     public final <T> T of(T... array) {
+        if (array.length == 0) {
+            return null;
+        } else if (array.length == 1) {
+            return array[0];
+        }
         return array[random.nextInt(array.length - 1)];
     }
 
     /**
      * Returns a random element from a list
      *
      * @param list the list to pick from
-     * @param <T> the type of object to return
+     * @param <T>  the type of object to return
      * @return a random element
```

Comment:
`@return a random element from the list or NULL if the list is NULL or empty` - or something like that.
Similarly in other `of()` methods.